libfwupdplugin/fu-intel-thunderbolt-nvm.c

```diff
@@ -448,11 +447,10 @@ fu_intel_thunderbolt_nvm_parse(FuFirmware *firmware,
 			   {0x1137, 4, FU_INTEL_THUNDERBOLT_NVM_FAMILY_MAPLE_RIDGE, 2},
 			   {0}};
 	g_autoptr(FuFirmware) img_payload = fu_firmware_new();
-	g_autoptr(GInputStream) stream_payload = NULL;
 	gsize streamsz = 0;
 
 	/* add this straight away */
-	priv->sections[FU_INTEL_THUNDERBOLT_NVM_SECTION_DIGITAL] = offset;
+	priv->sections[FU_INTEL_THUNDERBOLT_NVM_SECTION_DIGITAL] = 0x0;
```

Comment:
thunderbolt firmware (the binary) used to learn the starting offset for the digital section to take useful data. After this PR is applied, the offset discovered by https://github.com/fwupd/fwupd/pull/7887/files#diff-f7ea10c3bd60b915875d9126ab00ca7e32c2d0247b2eb9f77a91f09de157d38fL74 isn't being well handled.

I'm seeing an issue "Unknown NVM family" while parsing a stable gr firmware because the `device id`, etc. was wrong.

```
$ fwupdtool firmware-parse ~/Downloads/gorshenridge.bin -vv
...
Idle…: 0%
14:55:08.801 FuEngine             resetting update motd timeout
71
14:55:19.441 FuFirmware           detected digital section begins at 0x4000
unknown NVM family
14:55:19.443 FuPlugin             finalize(modem_manager)
14:55:19.443 FuPlugin             finalize(flashrom)
14:55:19.443 FuPluginTest         destroy
```

This issue is reproducible with WD22's firmware bundle that is available on public remote.

Reply:
> This issue is reproducible with WD22's firmware bundle that is available on public remote.

In a related note; can you try to record emulation data for WD22 again now?  I think you should be able to record the whole bundle and then that will ensure we regression test every commit.

Reply:
Good catch! https://github.com/fwupd/fwupd/pull/7906 should fix things. I missed it as FuIntelThunderboltNvm was being _weird_.